### PR TITLE
Check that email is verified, or filter roles

### PR DESCRIPTION
**Pull request by fredrikved**

### CODE REVIEW

File: backend/user/middleware.go [hunk 1]
```diff
@@ -31,6 +31,9 @@ const (
 	RoleNonBCCMember = "non-bcc-members"
 )
 
+// DefaultRoles are default roles in the system.
+var DefaultRoles = []string{RolePublic, RoleRegistered, RoleBCCMember, RoleNonBCCMember}
```
Comment (andreasgan) [inline backend/user/middleware.go:35]: const?

File: backend/user/middleware.go [hunk 2]
```diff
@@ -228,7 +234,13 @@ func NewUserMiddleware(queries *sqlc.Queries, remoteCache *remotecache.Client, l
 				roles = append(roles, userRoles...)
 			}
 
-			u.Roles = roles
+			if !u.EmailVerified {
```
Comment (andreasgan) [inline backend/user/middleware.go:237]: Any reason not to skip the "GetRolesForEmail" step instead of doing this filtering after?

**Review by andreasgan — CHANGES_REQUESTED** [review]

File: backend/user/middleware.go (hunk 1 above)
Comment (fredrikved) [reply backend/user/middleware.go:35]: arrays cant be consts, as they are in nature dynamic

Comment (andreasgan) [reply backend/user/middleware.go]: oh ok.
but you can make fixed size arrays with [...]
https://stackoverflow.com/questions/13137463/declare-a-constant-array

**Review by andreasgan — APPROVED** [review]

**Pull request merged**